### ISSUE TITLE
1\4

### ISSUE THREAD

**Issue by AndreiKapranov** [AndreiKapranov/Hello-World#5]
Раньше отрицательные числа представляли при помощи обратного кода,но там есть минус ноль.Поэтому перешли на дополнительный код.В обратном коде -0 это 1111 1111,а в дополнительном 1111 1111 это -1.Получается типа всё сдвинулось на единицу,поэтому 128 в однобайтовом формате нету. Потому что значение 1000 0000 уже занято под "-128".Я так это понимаю.Могу ошибаться.

**Comment by vanjuxin-dju** [comment]
Верно. Ну и мораль этого задания сводится к тому, что у переменных есть крайние значения, особенно это полезно в вычислениях больших объёмов данных, где например переменной типа _int_ может быть маловато, нужен тип _long_ (для целых чисел) или _double_ (для вещественных чисел), занимающий 8 байт.
Есть замечание к коду. В файле _Application1I4.java_ в 10 строке ты написал:
`c = ++c;` 
Дело в том, что операция `++c` уже выполняет присвоение в переменную `c`, поэтому в этой строке достаточно просто написать `++c` или `c++`.
И ещё: форматируй код, чтобы он был понятнее. В Eclipse есть такая комбинация _Ctrl+Shift+F_ для автоматического форматирования кода.
Задание 4 первого модуля засчитано, модуль завершён.
